src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaTests.cs

```diff
@@ -149,19 +140,16 @@ public void GenerateExportSecretKeySignAndVerify(SlhDsaAlgorithm algorithm)
 
             using (SlhDsa slhDsa = GenerateKey(algorithm))
             {
-                signature = new byte[algorithm.SignatureSizeInBytes];
-                Assert.Equal(signature.Length, slhDsa.SignData(data, signature));
-
-                secretKey = new byte[algorithm.SecretKeySizeInBytes];
-                Assert.Equal(secretKey.Length, slhDsa.ExportSlhDsaSecretKey(secretKey));
+                signature = slhDsa.SignData(data);
+                secretKey = slhDsa.ExportSlhDsaSecretKey();
             }
 
             using (SlhDsa slhDsa = ImportSlhDsaSecretKey(algorithm, secretKey))
             {
                 ExerciseSuccessfulVerify(slhDsa, data, signature, []);
 
                 signature.AsSpan().Clear();
-                Assert.Equal(signature.Length, slhDsa.SignData(data, signature));
+                slhDsa.SignData(data, signature, []);
```

Comment:
This might be something we can talk about in API review. Note that when `data` and `signature` are typed as arrays, `slhDsa.SignData(data, signature)` resolves to the `byte[] SignData(byte[] data, byte[]? context = null)` overload (2nd argument is treated as the context). If the user did what I did in the latest iteration here and changes from `sig = slhDsa.SignData(data)` to `slhDsa.SignData(data, sig)` they'll get a surprise at runtime.

Reply:
Ohhh yeah, ouch.